libraries/AC_PrecLand/AC_PrecLand_SITL.cpp

```diff
@@ -3,6 +3,8 @@
 
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
 
+#if AC_PRECLAND_SITL_ENABLED
```

Comment:
AC_PRECLAND_SITL_ENABLED should be true only if ` CONFIG_HAL_BOARD == HAL_BOARD_SITL` is also true, and that should be in the new `_config.h` file.  So we shouldn't have `#if CONFIG_HAL_BOARD == HAL_BOARD_SITL` here